openfisca_core/scripts/__init__.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+
+import sys
+import importlib
+
+
+TAX_BENEFIT_SYSTEM_OPTIONS = {
+    'country_package': {
+        'short': 'c',
+        'help': 'country package to use. If not provided, an automatic detection will be attempted by scanning the python packages installed in your environment which name contains the word "openfisca".'
+        },
+    'extensions': {
+        'short': 'e',
+        'help': 'extensions to load',
+        'nargs': '*'
+        },
+    'reforms': {
+        'short': 'r',
+        'help': 'reforms to apply to the country package',
+        'nargs': '*'
+        }
+    }
+
+
+def add_tax_benefit_system_arguments(parser):
+    for option, properties in TAX_BENEFIT_SYSTEM_OPTIONS.iteritems():
+        parser.add_argument(
+            '-{}'.format(properties['short']),
+            '--{}'.format(option),
+            action = 'store',
+            help = properties['help'],
+            nargs = properties.get('nargs')
+            )
+
+    return parser
+
+
+def build_tax_benefit_sytem(country_package_name, extensions, reforms):
+    if country_package_name:
+        try:
+            country_package = importlib.import_module(country_package_name)
+            assert hasattr(country_package, 'CountryTaxBenefitSystem')
+        except:
+            print('ERROR: `{}` does not seem to be a valid Openfisca country package.'.format(country_package_name))
+            sys.exit(1)
+    else:
+        country_package_name = detect_country_package()
+        country_package = importlib.import_module(country_package_name)
+
+    tax_benefit_system = country_package.CountryTaxBenefitSystem()
+
+    if extensions:
+        for extension in extensions:
+            tax_benefit_system.load_extension(extension)
+
+    if reforms:
+        for reform_path in reforms:
+            try:
+                [reform_package, reform_name] = reform_path.rsplit('.', 1)
+                reform_module = importlib.import_module(reform_package)
+                reform = getattr(reform_module, reform_name)
+                tax_benefit_system = reform(tax_benefit_system)
+            except:
+                print('ERROR: `{}` does not seem to be a valid Openfisca reform for `{}`.'.format(reform_path, country_package.__name__))
+                raise
+
+    return tax_benefit_system
+
+
+def detect_country_package():
+    import pkgutil
+    from importlib import import_module
+
+    installed_country_packages = []
+
+    for module_description in pkgutil.iter_modules():
+        module_name = module_description[1]
+        if module_name.lower().find('openfisca') >= 0:
+            module = import_module(module_name)
+            if hasattr(module, 'CountryTaxBenefitSystem'):
+                installed_country_packages.append(module_name)
+
+    if len(installed_country_packages) == 0:
```

Comment:
Python idiom: use `if not installed_country_packages`

💭 Personally I'm against this idiom, since it relies on automatic casting from lists to booleans!

What to do when there is a bad idiom? 🤔